diuf/sudoku/solver/rules/TurbotFishHint.java

```diff
@@ -134,28 +134,28 @@ public String getName() {
 	
     @Override
     public String getShortName() {
-        Class<? extends Grid.Region> region1 = baseSet.getClass();
-        Class<? extends Grid.Region> region2 = coverSet.getClass();
-        if (region1 == Grid.Row.class) {
-            if (region2 == Grid.Row.class)
+        int region1 = baseSet.getRegionTypeIndex();
+        int region2 = coverSet.getRegionTypeIndex();
+        if (region1 == 1) {
+            if (region2 == 1)
                 return "Sky";
             else
-				if (region2 == Grid.Column.class)
+				if (region2 == 2)
 					return "2SK";
 				else
 					return "TF";
         }
 		else {
-			if (region1 == Grid.Column.class)
-				if (region2 == Grid.Row.class)
+			if (region1 == 2)
+				if (region2 == 1)
 					return "2SK";
 				else
-					if (region2 == Grid.Column.class)
+					if (region2 == 2)
 						return "Sky";
 					else 
                 return "TF";
 			else
-				if (region2 == Grid.Block.class) 
+				if (region2 == 0) 
 					return "GXW";
 				else
 					return "TF";
```

Comment:
Thanks …
I was just playing with TurbotFish.java and I ended up adding Empty Rectangle and also to the similar ThreeStrongLinks.java I have already modified the local version of TurbotFishHint.java but I'll change it to accommodate your changes

Reply:
I have manually re-aligned these because the file was already being updated with ER changes but it is now merged successfully in Master. I also used the same approach for the extended version in ThreeStrongLinksHint.java